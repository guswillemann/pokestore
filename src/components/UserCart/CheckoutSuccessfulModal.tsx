import styled from 'styled-components';
import useModal from '../../contexts/ModalContext';
import { getThemeColor } from '../../theme/utils';

const CheckoutSuccessfulModalWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  width: 40rem;
  height: 30rem;
  position: relative;
  
  background-color: ${getThemeColor('primary')};
  color: white;

  border-radius: ${({ theme }) => theme.borderRadius};
  
  button {
    position: absolute;
    top: 1rem;
    right: 1rem;

    width: 3rem;

    background-color: transparent;
    border: none;
    cursor: pointer;

    filter: invert();

    &:hover {
      opacity: 0.8;
    }
  }

  h2 {
    margin-bottom: 2rem;
  }

  p {
    margin-top: 2rem;
  }
`;

export default function CheckoutSuccessfulModal({ CheckoutTotalPrice }: { CheckoutTotalPrice: string }) {
  const { toggleModal } = useModal();

  return (
    <CheckoutSuccessfulModalWrapper>
      <button type="button" onClick={toggleModal}>
        <img src="/images/x-icon.svg" alt="X" aria-label="Fechar modal" />
      </button>
      <h2>Obrigado !!!</h2>
      <p>VocÃª ganhou de volta</p>
      <p>{`R$ ${CheckoutTotalPrice}`}</p>
    </CheckoutSuccessfulModalWrapper>
  );
}