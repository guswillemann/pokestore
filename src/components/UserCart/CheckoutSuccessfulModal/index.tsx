import useModal from '../../../contexts/ModalContext';
import CheckoutSuccessfulModalWrapper from './styles';

export default function CheckoutSuccessfulModal({ CheckoutTotalPrice }: { CheckoutTotalPrice: string }) {
  const { toggleModal } = useModal();

  return (
    <CheckoutSuccessfulModalWrapper data-testid="checkout-modal">
      <button type="button" onClick={toggleModal}>
        <img src="/images/x-icon.svg" alt="X" aria-label="Close modal" />
      </button>
      <h2>Thank you !!!</h2>
      <p>Total price for your order:</p>
      <p>{`$${CheckoutTotalPrice}`}</p>
    </CheckoutSuccessfulModalWrapper>
  );
}