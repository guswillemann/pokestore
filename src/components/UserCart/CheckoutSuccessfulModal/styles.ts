import styled from 'styled-components';
import { getThemeColor } from '../../../theme/utils';

const CheckoutSuccessfulModalWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  width: 40rem;
  height: 30rem;
  position: relative;
  
  background-color: ${getThemeColor('primary')};
  color: white;

  border-radius: ${({ theme }) => theme.borderRadius};
  
  button {
    position: absolute;
    top: 1rem;
    right: 1rem;

    width: 3rem;

    background-color: transparent;
    border: none;
    cursor: pointer;

    filter: invert();

    &:hover {
      opacity: 0.8;
    }
  }

  h2 {
    margin-bottom: 2rem;
  }

  p {
    margin-top: 2rem;
  }
`;

export default CheckoutSuccessfulModalWrapper;
