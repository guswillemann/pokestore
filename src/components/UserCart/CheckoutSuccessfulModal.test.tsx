import { render, screen } from '../../../test/test-utils';
import CheckoutSuccessfulModal from './CheckoutSuccessfulModal';

describe('<CheckoutSuccessfulModal />', () => {
  it('should mathc the snapshot', () => {
    render(<CheckoutSuccessfulModal CheckoutTotalPrice={'100'} />)
    const modal = screen.getByTestId('checkout-modal');
    expect(modal).toMatchSnapshot();
  });
});