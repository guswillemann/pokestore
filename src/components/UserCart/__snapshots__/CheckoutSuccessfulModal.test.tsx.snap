// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckoutSuccessfulModal /> should mathc the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 40rem;
  height: 30rem;
  position: relative;
  background-color: darkred;
  color: white;
  border-radius: 8px;
}

.c0 button {
  position: absolute;
  top: 1rem;
  right: 1rem;
  width: 3rem;
  background-color: transparent;
  border: none;
  cursor: pointer;
  -webkit-filter: invert();
  filter: invert();
}

.c0 button:hover {
  opacity: 0.8;
}

.c0 h2 {
  margin-bottom: 2rem;
}

.c0 p {
  margin-top: 2rem;
}

<div
  class="c0"
  data-testid="checkout-modal"
>
  <button
    type="button"
  >
    <img
      alt="X"
      aria-label="Fechar modal"
      src="/images/x-icon.svg"
    />
  </button>
  <h2>
    Obrigado !!!
  </h2>
  <p>
    VocÃª ganhou de volta
  </p>
  <p>
    R$ 100
  </p>
</div>
`;
